beets/ui/commands.py

```diff
@@ -801,6 +845,48 @@ def should_resume(self, path):
                            "was interrupted. Resume (Y/n)?"
                            .format(displayable_path(path)))
 
+    def _get_plugin_choices(self, task):
+        """Get the extra choices appended to the plugins to the ui prompt.
+
+        The `before_choose_candidate` event is sent to the plugins, with
+        session and task as its parameters. Plugins are responsible for
+        checking the right conditions and returning a list of `PromptChoice`s,
+        which is flattened and checked for conflicts.
+
+        Raises `ValueError` if two of the choices have the same short letter.
+
+        Returns a list of `PromptChoice`s.
+        """
+        # Send the before_choose_candidate event and flatten list.
+        extra_choices = list(chain(*plugins.send('before_choose_candidate',
+                                                 session=self, task=task)))
+        # Take into account default options, for duplicate checking.
+        all_choices = [PromptChoice('a', 'Apply', None),
+                       PromptChoice('s', 'Skip', None),
+                       PromptChoice('u', 'Use as-is', None),
+                       PromptChoice('t', 'as Tracks', None),
+                       PromptChoice('g', 'Group albums', None),
+                       PromptChoice('e', 'Enter search', None),
+                       PromptChoice('i', 'enter Id', None),
+                       PromptChoice('b', 'aBort', None)] +\
+            extra_choices
+
+        short_letters = [c.short for c in all_choices]
+        if len(short_letters) != len(set(short_letters)):
+            # Duplicate short letter has been found.
+            duplicates = [i for i, count in Counter(short_letters).items()
+                          if count > 1]
+            for short in duplicates:
+                # Keep the first of the choices, removing the rest.
+                dup_choices = [c for c in all_choices if c.short == short]
+                for c in dup_choices[1:]:
+                    log.warn(u"Prompt choice '{0}' removed due to conflict "
+                             u"with '{1}' (short letter: '{2}')",
+                             c.long, dup_choices[0].long, c.short)
+                    extra_choices.remove(c)
```

Comment:
The conflict resolution makes sure that no "core" choices are removed, as they are at the beginning of `all_choices`. No real effort is made on deciding which choice to keep if they are all from plugins - "first come, first served" of sorts.